### ISSUE TITLE
[NumberInputField] ArrowLeft & ArrowRight doesn't work

### ISSUE THREAD

**Issue by mohammed-adil** [chakra-ui/chakra-ui#711]
**Describe the bug**

In NumberInputField component `ArrowLeft` & `ArrowRight` doesn't work.

**Expected Behavior**

I'm not sure if it's intended to work like this, but I'd really like to navigate through the entered data using keyboard arrow left & right keys

**To Reproduce**
- Use `NumberInputField` component:
1. Enter any number in the field
2. Click on `ArrowLeft` or `ArrowRight` to navigate through entered data
3. We cannot navigate between the entered data

**Suggested solution(s)**

We could allow keyboard keys `37` (ArrowLeft) & `39` (ArrowRight) for the navigation
Also I found [this](https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/useNumberInput/utils.js#L4-L10), where chakra only allows [numeric keypad](https://github.com/chakra-ui/chakra-ui/issues/254)

**Desktop (please complete the following information):**
 - OS: macOS Catalina (10.15.3)
 - Browser: Chrome

**Comment by tristandubbeld** [comment]
Hey @mohammed-adil, I was looking for something to fix so I checked the issue and it seems that it has been fixed in the `dev-ts` branch.

[This line of code](https://github.com/chakra-ui/chakra-ui/blob/dev-ts/packages/number-input/src/NumberInput.utils.ts#L24) checks the length of `event.key` (which is either `ArrowLeft` or `ArrowRight` for this use case). When that length is not equal to 1, it returns as true as a valid numeric keyboard event and does not prevent default anymore.
